src/context/auth/authContext.js

```diff
@@ -0,0 +1,68 @@
+import { createContext, useContext, useReducer } from "react";
+import { loginApi, signupApi } from "../../services/auth_services";
+import { reducer } from "./authReducer";
+
+const AuthContext = createContext();
+const AuthContextProvider = ({ children }) => {
+  const user = JSON.parse(localStorage.getItem("user"));
+  const intialState = user ?? {
+    isUserloggedIn: false,
+    token: "",
+    user: null,
+  };
+
+  const [{ isUserloggedIn, token }, dispatch] = useReducer(
+    reducer,
+    intialState
+  );
+  const signup = async ({ email, password, firstName, lastName }) => {
+    const { data, success } = await signupApi({
+      firstName,
+      lastName,
+      email,
+      password,
+    });
+    if (success) {
+      dispatch({
+        type: "UPDATE_USER",
+        payload: {
+          isUserloggedIn: true,
+          token: data.encodedToken,
```

Comment:
Token should be stored in localstorage and not in context

Because context data would go on hard refresh, but localstorage data would persist

Reply:
It will persist the data as it is stored in localstorage . I am just trying to store this token and use it everywhere.